components/camel-fhir/camel-fhir-component/src/main/java/org/apache/camel/component/fhir/FhirComponent.java

```diff
@@ -45,18 +45,28 @@ protected FhirApiName getApiName(String apiNameStr) {
         return getCamelContext().getTypeConverter().convertTo(FhirApiName.class, apiNameStr);
     }
 
+    @Override
+    protected void afterPropertiesSet(FhirConfiguration endpointConfiguration) {
+        // ensure a client is set on the config
+        if (endpointConfiguration.getClient() == null && configuration.getClient() == null) {
+            if (configuration != null) {
```

Comment:
I apologize if I am missing context, but by looking at it, it seems that the `configuration` variable could be null. In which case, it would throw an NPE at line 51. Maybe it should fail right away by adding a pre-check with `ObjectHelper::notNull` ?

Reply:
many thanks @orpiske ! actually no test were failing due to NPE, but I noticed that all tests were using a custom FhirComponent, if someone use the default one, NPE will happend, I just created a new test for this usecase FhirSimpleIT.java

Reply:
Thanks for double-checking.